packages/vite/src/utils/generator-utils.ts

```diff
@@ -356,11 +359,15 @@ export function createOrEditViteConfig(
   projectAlreadyHasViteTargets?: TargetFlags,
   vitestFileName?: boolean
 ) {
-  const { root: projectRoot } = readProjectConfiguration(tree, options.project);
+  const { root: projectRoot, projectType } = readProjectConfiguration(
+    tree,
+    options.project
+  );
 
+  const extension = options.useEsmExtension ? 'mts' : 'ts';
   const viteConfigPath = vitestFileName
```

Comment:
Vite commonjs support is deprecated and angular vitest plugin supports ESM-only, that is why we have to use `.mts` if package type is not `module`.